sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2.scala

```diff
@@ -179,3 +192,24 @@ class InternalRowDataWriter(rowWriter: DataWriter[Row], encoder: ExpressionEncod
 
   override def abort(): Unit = rowWriter.abort()
 }
+
+/**
+ * Collects commit progress on writers.
+*/
+trait StreamWriterProgressCollector {
```

Comment:
This is a kinda weird interface. Can't we just create a StreamWriterCommitProgress(numOutputRows) instead of round-tripping through here?

Reply:
You are right. I havent realized that I can access WriteToDataSourceV2Exec directly. Please have a look now